week1/Q2_test.py

```diff
@@ -0,0 +1,27 @@
+import Q2
+import unittest
+
+
+class SLLTest(unittest.TestCase):
+
+	def testBaseCases(self):
+		self.assertEqual(9, Q2.get_kth_to_last(L, 0))
+		self.assertEqual(0, Q2.get_kth_to_last(L, 9))
+
+	def testKnownCases(self):
+		self.assertEqual(5, Q2.get_kth_to_last(L, 4))
+		self.assertEqual(3, Q2.get_kth_to_last(L, 6))
+ 
+	def testFalseCases(self):
+		self.assertEqual(False, Q2.get_kth_to_last(L, -1))
+		self.assertEqual(False, Q2.get_kth_to_last(L_empty, 0))
+		self.assertEqual(False, Q2.get_kth_to_last(L, 10))
+
+ 
+if __name__ == '__main__':
+	L = Q2.LinkedList()
```

Comment:
Creating the `L` value here is not ideal because:

* it can be modified
* it is not clear in context (when you write the test)

Instead, you can create a `setUp()` method on your test itself that can be used to store a variable as a field on the test. This variable is re-created each time, so that one test cannot affect the other.